pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -436,7 +460,13 @@ func modifyOutboundRouteConfig(push *model.PushContext, in *plugin.InputParams,
 		default:
 			log.Warn("Unknown cluster type in mixer#OnOutboundRouteConfiguration")
 		}
-	case *route.Route_Redirect, *route.Route_DirectResponse:
+	case *route.Route_DirectResponse:
+		if virtualHostname == "block_all" {
+			hostname := model.Hostname(util.BlackHoleCluster)
+			attrs := addDestinationServiceAttributes(make(attributes), push, hostname)
+			addFilterConfigToRoute(in, httpRoute, attrs, isXDSMarshalingToAnyEnabled)
+		}
+	case *route.Route_Redirect:
```

Comment:
Needs a TODO or an explanation why this was left-out.

Reply:
will add explanation.